modules/kubernetes-addons/main.tf

```diff
@@ -152,15 +152,15 @@ module "metrics_server" {
 }
 
 module "prometheus" {
-  count       = var.enable_prometheus ? 1 : 0
-  source      = "./prometheus"
-  helm_config = var.prometheus_helm_config
-
+  count          = var.enable_prometheus ? 1 : 0
+  source         = "./prometheus"
+  eks_cluster_id = var.eks_cluster_id
+  helm_config    = var.prometheus_helm_config
   #AWS Managed Prometheus Workspace
-  amazon_prometheus_workspace_id           = var.amazon_prometheus_workspace_id
-  amazon_prometheus_ingest_iam_role_arn    = var.amazon_prometheus_ingest_iam_role_arn
-  amazon_prometheus_ingest_service_account = var.amazon_prometheus_ingest_service_account
-  manage_via_gitops                        = var.argocd_manage_add_ons
+  enable_amp_for_prometheus      = var.enable_amp_for_prometheus
```

Comment:
I don't think this variable is necessary. If `amazon_prometheus_workspace_id` is not null, then we can assume that `enable_amp_for_prometheus` would be true right? 

Also consider changing `amazon_prometheus_workspace_id` to `amp_workspace_id`

Reply:
We have three scenarios here...

1. Users passing existing AMP workspace ID to integrate with Prometheus (This will integrate existing AMP WS with Prometheus.)
`enable_prometheus=true` &&`enable_amp_for_prometheus = true` && `amazon_prometheus_workspace_id=something` 
2. Users enabling Prometheus only (This should ignore AMP config)
`enable_prometheus=true` && `enable_amp_for_prometheus = false`(default)
3.  Users don't pass AMP workspace but they enable AMP with Prometheus (This will create AMP and integrate with Prometheus. amazon_prometheus_workspace_id is null here)
    `enable_amp_for_prometheus = true` && `enable_prometheus=true` && `amazon_prometheus_workspace_id=null` 

Overall `enable_amp_for_prometheus = true` is critical to integrate the AMP workspace ID with Prometheus or not.
`amazon_prometheus_workspace_id` is used  to check whether to create a new AMP Woekspace or not.